src/tasks/LibraryBuilder/Templates/autoinit.c

```diff
@@ -18,6 +18,33 @@
 
 #include "library-builder.h"
 
+#if defined(BUNDLED_ASSEMBLIES)
+
+static void
+cleanup_runtime_config (MonovmRuntimeConfigArguments *args, void *user_data)
+{
+    free (args);
+    free (user_data);
+}
+
+static void
+initialize_runtimeconfig ()
+{
+    MonovmRuntimeConfigArguments *arg = (MonovmRuntimeConfigArguments *)malloc (sizeof (MonovmRuntimeConfigArguments));
+    if (!arg)
+        LOG_ERROR ("Out of memory.\n");
+
+    arg->kind = 1;
+    const char *data;
+    int data_len;
+    mono_get_bundled_assembly ("runtimeconfig.bin", &data, &data_len);
```

Comment:
This API should probably be named something different, since you clearly don't only use it for bundled assemblies.

Reply:
modified to be `mono_add_bundled_resource`, and takes in `BundledResourceType` to classify the `BundledResource` that was added to the hash table.